autogluon/task/tabular_prediction/predictor.py

```diff
@@ -386,6 +388,43 @@ def feature_importance(self, model=None, dataset=None, features=None, raw=True,
 
         return self._learner.get_feature_importance(model=model, X=dataset, features=features, raw=raw, subsample_size=subsample_size, silent=silent)
 
+    def refit_full(self, model=None):
+        """
+        Optimizes a model's inference time by collapsing bagged ensembles into a single model fit on all of the training data.
+        For bagged models, this process will typically result in a slight accuracy reduction and a large inference speedup.
+            The inference speedup will generally be between 10-200x faster than the original bagged ensemble model.
+            The runtime is generally 10% or less of the original fit runtime.
+        This function can also be called on non-bagged models for a slight accuracy increase and no change to inference time.
+            In this instance, the runtime will be approximately equal to the original fit runtime.
+        This process does not alter the original models, but instead adds additional models.
+        Models produced by this process will not have validation scores, as they use all of the data for training.
+        Therefore, it is up to the user to determine if the models are of sufficient quality by including test data in `predictor.leaderboard(dataset=test_data)`.
+
+        Parameters
+        ----------
+        model : str, default = None
+            Model to refit. If None then all models are refitted.
```

Comment:
should we refit the best_model by default instead, so this runs faster?  
Eg. if the WeightedEnsemble only depends on models #1 and #2, it probably is unnecessary to refit model #3 by default right?

Reply:
Good point, however in many cases a user may be using this functionality to try to find a good accuracy/latency tradeoff, which isn't necessarily the 'best model'. This is already quite fast relative to the normal training procedure. What we could do is add an option such as model='best', model='all', to enable those use-cases. I still think the default should be model='all' given how cheap it is to compute.

Reply:
Good point re accuracy/latency, I like the idea of offering model='best', model='all' options, with model='all' as the default.